code/game/objects/structures/safe.dm

```diff
@@ -40,107 +55,137 @@ FLOOR SAFES
 			I.loc = src
 
 
-/obj/structure/safe/proc/check_unlocked(mob/user, canhear)
-	if(user && canhear)
-		if(tumbler_1_pos == tumbler_1_open)
-			to_chat(user, "<span class='italics'>You hear a [pick("tonk", "krunk", "plunk")] from [src].</span>")
-		if(tumbler_2_pos == tumbler_2_open)
-			to_chat(user, "<span class='italics'>You hear a [pick("tink", "krink", "plink")] from [src].</span>")
-	if(tumbler_1_pos == tumbler_1_open && tumbler_2_pos == tumbler_2_open)
-		if(user) visible_message("<i><b>[pick("Spring", "Sprang", "Sproing", "Clunk", "Krunk")]!</b></i>")
+/obj/structure/safe/proc/check_unlocked()
+	if(tumbler_1_pos == tumbler_1_open && tumbler_2_pos == tumbler_2_open && dial == open_pos)
+		unlocked = 1
 		return 1
+	unlocked = 0
 	return 0
 
-
-/obj/structure/safe/proc/decrement(num)
-	num -= 1
-	if(num < 0)
-		num = 71
-	return num
-
-
-/obj/structure/safe/proc/increment(num)
-	num += 1
-	if(num > 71)
-		num = 0
-	return num
-
+/obj/structure/safe/proc/make_noise(tum1_turns, tum2_turns, mob/user, canhear)
+	if(user && canhear)
+		var/tum1_tmp = tum1_turns
+		var/tum2_tmp = tum2_turns
+		while(tum1_tmp >= 1 || tum2_tmp >= 1)
+			if(tum1_tmp >= 1)
+				to_chat(user, "<span class='italics'>You hear a [pick("clack", "scrape", "clank")] from [src].</span>")
+				tum1_tmp = tum1_tmp - rand(4, 8)
+			if(tum2_tmp >= 1)
+				to_chat(user, "<span class='italics'>You hear a [pick("click", "chink", "clink")] from [src].</span>")
+				tum2_tmp = tum2_tmp - rand(4, 8)
+			if(tumbler_1_pos == tumbler_1_open && tum1_turns == 1) // You cant hear tumblers if you spin fast!
+				to_chat(user, "<span class='italics'>You hear a [pick("tonk", "krunk", "plunk")] from [src].</span>")
+			if(tumbler_2_pos == tumbler_2_open && tum1_turns == 1 && tum2_turns == 1) // You cant hear tumblers if you spin fast!
+				to_chat(user, "<span class='italics'>You hear a [pick("tink", "krink", "plink")] from [src].</span>")
+	if(unlocked)
+		if(user) visible_message("<i><b>[pick("Spring", "Sprang", "Sproing", "Clunk", "Krunk")]!</b></i>")
 
 /obj/structure/safe/update_icon()
 	if(open)
 		icon_state = "[initial(icon_state)]-open"
 	else
 		icon_state = initial(icon_state)
 
-
 /obj/structure/safe/attack_hand(mob/user)

```

Comment:
I may be missing something here, but is there a way to put items back into the safe?

Reply:
Ah ok